src/java/com/puppetlabs/ssl_utils/SSLUtils.java

```diff
@@ -438,6 +445,60 @@ public static X509CRL revokeMultiple(X509CRL crl,
         return buildCRL(crl, issuerPrivateKey, issuerPublicKey, builder);
     }
 
+    /**
+     * Given a certificate chain and CRLs, validate CRLS and all certs in the chain.
+     *
+     * @param certs The certificate chain to validate
+     * @param crls The CRL chain to validate
+     * @throws CertificateException
+     * @throws CertPathValidatorException
+     * @throws IOException
+     * @throws InvalidAlgorithmParameterException
+     * @throws KeyStoreException
+     * @throws NoSuchAlgorithmException
+     * @throws NoSuchProviderException
+     */
+    public static void validateCertChain(List<X509Certificate> certs,
+                                       List<X509CRL> crls)
+        throws CertificateException, CertPathValidatorException, IOException,
+               InvalidAlgorithmParameterException, KeyStoreException,
+               NoSuchAlgorithmException, NoSuchProviderException {
+        // there are bouncycastle versions of CertificateFactory and CertPath
+        // that I haven't looked into - not sure when using bouncycastle vs. java
+        // makes sense
```

Comment:
I think typically using the Java apis were available are more robust to changing BC versions or modes like FIPS vs non-FIPs. Though that idea kinda flies in the face of the NoSuchAlgorithmException you're getting below.

Reply:
I was wondering about that, but that error is getting thrown by the `CertPathValidator` class, which doesn't seem to have a corresponding BC class, afaict.

Reply:
so I guess that's why